adb-shell: add page (#4287)
tldr-pages/tldr: pages/common/adb-shell.md

# adb shell

> > Android Debug Bridge Shell: Run remote shell command with an Android emulator instance or connected Android devices.

Reviewer: Perhaps this would be more consistent with other existing pages?
```suggestion
> Android Debug Bridge Shell: Run remote shell commands on an Android emulator instance or connected Android devices.
```

> More information: <https://developer.android.com/studio/command-line/adb>.

> - Start a remote interactive shell in the emulator/device:

Reviewer: Perhaps this would be more consistent with other existing pages?
```suggestion
- Start a remote interactive shell on the emulator/device:
```

`adb shell`

> - Reset dangerous permissions to an application:

Reviewer: I'm not quite sure what you mean here - it's a little unclear. Do you mean `for` instead of `to` here?
```suggestion
- Reset dangerous permissions for an application:
```

`adb shell pm reset-permissions {{package}}`

> - Revoke a dangerous permission to an application:

Reviewer: Same here?
```suggestion
- Revoke a dangerous permission for an application:
```

`adb shell pm revoke {{package}} {{permission}}`

- Trigger a key event:

`adb shell input keyevent {{keycode}}`

> - Clear the data from an emulator/device's application:

Reviewer: Perhaps this will read better?
```suggestion
- Clear the data of an application on an emulator or device:
```

`adb shell pm clear {{package}}`

- Start an activity on emulator/device:

`adb shell am start -n {{package}}/{{activity}}`

> - Start home activity on emulator/device:

Reviewer: Perhaps this will read better?
```suggestion
- Start the home activity on an emulator or device:
```

Also, is this the main home screen, or the home screen of a specific application? If it's the latter, then perhaps something like this would be better?
```suggestion
- Start the home activity for an application on an emulator or device:
```

Finally, are there any long forms of the arguments used here? We generally prefer long forms of arguments in tldr pages.

Author: Is the Android home.

adb has only short forms for most arguments
